src/test/java/org/cactoos/io/JoinedTest.java

```diff
@@ -23,33 +23,97 @@
  */
 package org.cactoos.io;
 
+import java.io.DataInputStream;
 import java.io.IOException;
+import org.cactoos.bytes.HexOf;
+import org.cactoos.iterable.IterableOf;
 import org.cactoos.text.TextOf;
-import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsEqual;
 import org.junit.Test;
+import org.llorllale.cactoos.matchers.Assertion;
 import org.llorllale.cactoos.matchers.TextHasString;
 
 /**
  * Unit tests for {@link Joined}.
  * @since 0.36
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
 public final class JoinedTest {
     /**
      * Must join inputs in the given order.
      * @throws IOException If an error occurs
      */
     @Test
-    public void joinsOk() throws IOException {
-        MatcherAssert.assertThat(
+    public void joinsOk() {
+        new Assertion<>(
             "Cannot properly join inputs",
-            new TextOf(
+            () -> new TextOf(
                 new Joined(
                     new InputOf("first"),
                     new InputOf("second"),
                     new InputOf("third")
                 )
             ),
             new TextHasString("firstsecondthird")
-        );
+        ).affirm();
+    }
+
+    /**
+     * Must join inputs of the iterable in the given order.
+     */
+    @Test
+    public void fromIterable() {
+        new Assertion<>(
+            "Can't join iterable of inputs",
+            () -> new TextOf(
+                new Joined(
+                    new IterableOf<>(
+                        new InputOf("ab"),
+                        new InputOf("cde"),
+                        new InputOf("fghi")
+                    )
+                )
+            ),
+            new TextHasString("abcdefghi")
+        ).affirm();
+    }
+
+    /**
+     * Must join inputs of the iterable bytes.
+     * @checkstyle MagicNumberCheck (20 lines)
+     * @throws Exception If an error occurs
+     */
+    @Test
+    public void fromIterableInputOfBytes() throws Exception {
+        final byte[] result = new byte[6];
+        new DataInputStream(
+            new Joined(
+                new IterableOf<>(
+                    new InputOf(
+                        new BytesOf(
+                            (byte) 202,
+                            (byte) 254,
+                            (byte) 186,
+                            (byte) 190
+                        )
+                    ),
+                    new InputOf(
+                        new BytesOf(
+                            (byte) 222,
+                            (byte) 173
+                        )
+                    )
+                )
+            ).stream()
+        ).readFully(result);
+        new Assertion<>(
+            "The merged stream does not equal to the expected sum of 2",
+            () -> result,
+            new IsEqual<>(
+                new HexOf(
+                    new TextOf("CAFEBABEDEAD")
```

Comment:
@ilyakharlamov this is too hard to read. My human eyes can't correlate these ASCII letters with those bytes above. This test can be removed.

Reply:
@llorllale Sorry, but I disagree. We have similiar tests in `GzipInputTest` and `GzipOutputTest`. And is a important test. Please, don't delete it.

Reply:
@llorllale Sorry, but I disagree. We have similiar tests in `GzipInputTest` and `GzipOutputTest`. And is an important test. Please, don't delete it.

Reply:
@fabriciofx and they are totally unreadable, thanks for pointing them out to me. But `Joined` here does not concern itself with bytes - it operates on a higher plane of abstraction.

@ilyakharlamov can you leave a `@todo` (or open a new ticket - whichever way you prefer) to make those tests for `GzipInput` and `GzipOutput` readable?

Reply:
@llorllale Right.
@ilyakharlamov IMHO it's better open a new ticket.